include/picongpu/simulation/stage/FieldBackground.hpp

```diff
@@ -38,55 +45,255 @@ namespace picongpu
+                        , duplicateField(duplicateField)
+                        , isEnabled(FieldBackground::InfluenceParticlePusher)
+                    {
+                        if(isEnabled && duplicateField)
+                        {
+                            // Allocate a duplicate field buffer and copy the values
+                            DataConnector& dc = Environment<>::get().DataConnector();
+                            auto field = dc.get<Field>(Field::getName(), true);
+                            auto const& gridBuffer = field->getGridBuffer();
+                            duplicateBuffer = pmacc::makeDeepCopy(gridBuffer.getDeviceBuffer());
+                            dc.releaseData(Field::getName());
+                        }
+                    }
+
+                    /** Add the field background in the given area
+                     *
+                     * @tparam T_area area to operate on
+                     *
+                     * @param step index of time iteration
+                     */
+                    template<uint32_t T_area>
+                    void add(uint32_t const step) const
+                    {
+                        if(!isEnabled)
+                            return;
+                        DataConnector& dc = Environment<>::get().DataConnector();
+                        auto& field = *dc.get<Field>(Field::getName(), true);
+                        // Always add, conditionally make a copy first
+                        if(duplicateField)
+                        {
+                            auto& gridBuffer = field.getGridBuffer();
+                            duplicateBuffer->copyFrom(gridBuffer.getDeviceBuffer());
+                            __getTransactionEvent().waitForFinished();
+                        }
+                        apply<T_area>(step, pmacc::nvidia::functors::Add(), field);
+                        dc.releaseData(Field::getName());
+                    }
+
+                    /** Subtract the field background in the given area
+                     *
+                     * @tparam T_area area to operate on
+                     *
+                     * @param step index of time iteration
+                     */
+                    template<uint32_t T_area>
+                    void subtract(uint32_t const step) const
+                    {
+                        if(!isEnabled)
+                            return;
+                        DataConnector& dc = Environment<>::get().DataConnector();
+                        auto& field = *dc.get<Field>(Field::getName(), true);
+                        // Either restore from the pre-made copy or subtract
+                        if(duplicateField)
+                        {
+                            auto& gridBuffer = field.getGridBuffer();
+                            gridBuffer.getDeviceBuffer().copyFrom(*duplicateBuffer);
+                            __getTransactionEvent().waitForFinished();
```

Comment:
There is no need to wait for the event system.

Reply:
Fixed